robocrys/condense/structure.py

```diff
@@ -0,0 +1,344 @@
+"""
+This module defines a class for condensing structures into dict representations.
+"""
+
+from typing import Optional, Dict, Any, List, Tuple
+
+from pymatgen.analysis.dimensionality import get_structure_components
+from pymatgen.analysis.local_env import NearNeighbors, CrystalNN
+from pymatgen.core.structure import Structure
+from pymatgen.symmetry.analyzer import SpacegroupAnalyzer
+
+from robocrys import common_formulas
+from robocrys.condense.component import (get_sym_inequiv_components,
+                                         get_reconstructed_structure,
+                                         get_formula_from_components,
+                                         get_component_formula,
+                                         components_are_vdw_heterostructure,
+                                         get_vdw_heterostructure_information,
+                                         get_structure_inequiv_components,
+                                         Component)
+from robocrys.condense.mineral import MineralMatcher
+from robocrys.condense.molecule import MoleculeNamer
+from robocrys.condense.site import SiteAnalyzer
+
+
+class StructureCondenser(object):
+    """Class to transform a structure into an intermediate dict representation.
+
+    Args:
+        use_conventional_cell: Whether to always use the convention cell
+            representation of the structure.
+        near_neighbors: A ``NearNeighbors`` instance used to calculate the
+            bonding in the structure. For example, one of
+            :class:`pymatgen.analysis.local_env.CrystalNN`,
+            :class:`pymatgen.analysis.local_env.VoronoiNN`, etc. Defaults to
+            ``None``, in which case
+            :class:`pymatgen.analysis.local_env.CrystalNN` will be used.
+        mineral_matcher: A ``MineralMatcher`` instance. Defaults to ``None``
+            in which case the default ``MineralMatcher`` settings will be used.
+        use_symmetry: Whether to use symmetry to determine if structure
```

Comment:
Can you combine these two into one parameter?

e.g. symprec=None means don't use symmetry

(also for the CLI)